#Le codage binaire

##Rappel : Bibliothèque de référence Python
https://docs.python.org/fr/3/reference/index.html#reference-index

##Objectif du programme :
##Etape 1 :
Ecrire une boucle de 0 à 300 qui écrit la conversion en binaire d'un nombre.
On obtient en sortie :

0 sécrit en binaire  0
1 sécrit en binaire  1
2 sécrit en binaire  10
3 sécrit en binaire  11.....

In [8]:
#Etape 1 et 2:
for a in range(0 ,300 ):
    #convertir a en binaire et l'affecter à une nouvelle variable
    abin = bin(a)
    #Ecrire la phrase de synthese
    print (a, "s'écrit en binaire : ", abin[2:], "sur ", len(abin[2:]) , " bits")

0 s'écrit en binaire :  0 sur  1  bits
1 s'écrit en binaire :  1 sur  1  bits
2 s'écrit en binaire :  10 sur  2  bits
3 s'écrit en binaire :  11 sur  2  bits
4 s'écrit en binaire :  100 sur  3  bits
5 s'écrit en binaire :  101 sur  3  bits
6 s'écrit en binaire :  110 sur  3  bits
7 s'écrit en binaire :  111 sur  3  bits
8 s'écrit en binaire :  1000 sur  4  bits
9 s'écrit en binaire :  1001 sur  4  bits
10 s'écrit en binaire :  1010 sur  4  bits
11 s'écrit en binaire :  1011 sur  4  bits
12 s'écrit en binaire :  1100 sur  4  bits
13 s'écrit en binaire :  1101 sur  4  bits
14 s'écrit en binaire :  1110 sur  4  bits
15 s'écrit en binaire :  1111 sur  4  bits
16 s'écrit en binaire :  10000 sur  5  bits
17 s'écrit en binaire :  10001 sur  5  bits
18 s'écrit en binaire :  10010 sur  5  bits
19 s'écrit en binaire :  10011 sur  5  bits
20 s'écrit en binaire :  10100 sur  5  bits
21 s'écrit en binaire :  10101 sur  5  bits
22 s'écrit en binaire :  10110 sur  5  bits
23 s'écrit en binaire :  101

##Etape 2 :
Est ce que l'affichage est identique à ce qui est demandé  ?
Comment résoudre le problème ?

Aide : il faut enlever des caractères à la variable abin
http://www.python-simple.com/python-langage/chaines-de-caracteres-string.php 

##Etape 3 :
On veut enrichir la phrase de synthèse en calculant la longueur de la conversion binaire.
0 sécrit en binaire  0  sur  1  bits
1 sécrit en binaire  1  sur  1  bits
2 sécrit en binaire  10  sur  2  bits
3 sécrit en binaire  11  sur  2  bits ....

In [ ]:
#Etape 3:
for a in range( , ):
    #convertir a en binaire et l'affecter à une nouvelle variable
    abin = a
    #Ecrire la phrase de synthese
    print (a)

##Etape 4 :
A l'aide du programme que vous venez d'écrire, convertir en binaire les nombres entre 1 000 000 et 1 000 300

In [ ]:
#Etape 4:
for a in range( , ):
    #convertir a en binaire et l'affecter à une nouvelle variable
    abin = a
    #Ecrire la phrase de synthese
    print (a)

Est-ce encore lisible ?
Que peut on faire ?

Existe il d'autres représentation des nombres entiers en informatique ?

##Etape 5 :
Mettez en oeuvre une nouvelle méthode de codage XXXX pour des nombres de 0 à 300 puis de 1 000 000 à 1 000 300

In [ ]:
#Etape 5 : 
for a in range(0, 300):
    #convertir a en binaire et l'affecter à une nouvelle variable
    abin = a
    #Ecrire la phrase de synthese
    print (a)

##Etape 6 :
Reprenez votre programme et modifier la phrase de synthèse pour afficher côte à côté le binaire et le nouveau codage
exemple : 
15 sécrit en binaire  1111 et en XXXX  ?

In [ ]:
#Etape 6 : 
for a in range(0, 300):
    #convertir a en binaire et l'affecter à une nouvelle variable
    abin = a
    #Ecrire la phrase de synthese
    print (a)

Que remarquez vous ?